pkg/snapshot/snapshotter/snapshotter.go

```diff
@@ -351,11 +354,25 @@ func (ssr *Snapshotter) CollectEventsSincePrevSnapshot(stopCh <-chan struct{}) (
 	}
 	lastEtcdRevision := resp.Header.Revision
 
+	metrics.SnapshotRequired.With(prometheus.Labels{metrics.LabelKind: snapstore.SnapshotKindFull}).Set(0)
+	metrics.SnapshotRequired.With(prometheus.Labels{metrics.LabelKind: snapstore.SnapshotKindDelta}).Set(0)
+
+	// if etcd revision newer than latest full snapshot revision,
+	// set `required` metric for full snapshot to 1
+	if ssr.PrevFullSnapshot == nil || ssr.PrevFullSnapshot.LastRevision != lastEtcdRevision {
```

Comment:
We don't need this `if` block right? Check is already done on  next line.

Reply:
This check is for previous full snapshot, whereas next check is for previous snapshot (it can be full or delta). Even if there are no new events since previous delta snapshot, there could be new events since previous full snapshot (this is the case most of the time). So we need a separate check for previous full snapshot.